python/lancedb/table.py

```diff
@@ -373,15 +375,21 @@ def restore(self, version: int):
         3
         """
         max_ver = max([v["version"] for v in self._dataset.versions()])
+        if version is None:
+            version = self.version
+
         if version < 1 or version >= max_ver:
             raise ValueError(f"Invalid version {version}")
+        if version is None:
+            version = self.version
```

Comment:
Looks like you have this check twice (here and on 378).  Is it needed in both spots?

Reply:
forgive me father for i have sinned. On this day in the year 2023 of our lord, i committed the grave sin of over-reliance on github copilot.